### PR TITLE
tests: random: add test case for coverage

### PR DESCRIPTION
**Pull request by ghost**
Add test for random number generator APIs in userspace.

Signed-off-by: Lixin Guo <lixinx.guo@intel.com>

### CODE REVIEW

**Review by enjiamai — APPROVED** [review]

**Review by Zhaoningx — DISMISSED** [review]

**Comment by LixinGuoX** [comment]
Anyone available to review this one ?

**Review by LeiW000 — DISMISSED** [review]

**Comment by LeiW000** [comment]
@nashif , could you help to merge it?

**Comment by enjiamai** [comment]
Hi @LixinGuoX could you rebase it?

**Review by peter-mitsis — DISMISSED** [review]

File: tests/crypto/rand32/testcase.yaml [hunk 1]
```diff
@@ -16,3 +16,6 @@ tests:
     filter: CONFIG_ENTROPY_HAS_DRIVER
     tags: crypto entropy random security
     min_ram: 16
+  crypto.rand32.user:
+    tags: crypto random security userspace
+    min_ram: 16
```
Comment (chen-png) [inline tests/crypto/rand32/testcase.yaml:21]: maybe you should put 'CONFIG_TEST_USERSPACE=y' at here using extra_configs options, if you added it in prj.conf, it suits for all testcases in this test.

Comment (enjiamai) [reply tests/crypto/rand32/testcase.yaml:21]: The concern from Peng is right. And I check the coverage, it seems like by your test case doesn't cover rand32_handlers.c, could you please refine it again? Thanks!

**Review by enjiamai — CHANGES_REQUESTED** [review]
Please also develop the code coverage for rand32_hanlders.c also, thanks!

File: tests/crypto/rand32/testcase.yaml (hunk 1 above)
Comment (LixinGuoX) [reply tests/crypto/rand32/testcase.yaml]: Thank you for the suggestion. Yes, it is inappropriate to run the same case twice in kernel and userspace, so I split the case to two to run the new one in userspace. Is this appropriate?

File: tests/crypto/rand32/src/main.c [hunk 2]
```diff
@@ -114,10 +114,67 @@ void test_rand32(void)
 #endif /* CONFIG_CSPRING_ENABLED */
 }
 
+static void test_user_rand32(void)
+{
+#ifndef CONFIG_USERSPACE
+	ztest_test_skip();
+#endif /* CONFIG_USERSPACE */
```
Comment (LixinGuoX) [inline tests/crypto/rand32/src/main.c:121]: Any suite in testcase.yaml who doesn't support userspace will skip this case.

**Comment by LixinGuoX** [comment]
Lone time no update, so close it.

**Pull request closed without merging**